packages/gamut-styles/src/ColorMode.tsx

```diff
@@ -34,6 +35,21 @@ export const providerProps = variance.compose(
   space
 );
 
+export function useColorModes(): [
+  ColorModes,
+  ColorModeShape,
+  ColorModeConfig,
+  (color: Colors) => string
+] {
+  const { mode, modes, _getColorValue: getColorValue } = useTheme() || {};
+  return [mode, modes?.[mode], modes, getColorValue];
+}
+
+export function useCurrentMode(mode?: ColorModes) {
+  const [activeMode] = useColorModes();
+  return mode ?? activeMode;
+}
```

Comment:
Nit: there's no real need to call `useColorModes()` if it's just looking up from theme, is there?

```suggestion
export function useCurrentMode(mode?: ColorModes) {
  const theme = useTheme() ?? {};
  return mode ?? theme.mode;
}
```